edit-post/components/layout/index.js

```diff
@@ -112,26 +115,31 @@ function Layout( {
 				openedGeneralSidebar !== null && <GeneralSidebar
 					openedGeneralSidebar={ openedGeneralSidebar } />
 			}
+			{ layoutHasOpenSidebar && isMobileViewport && <MobileScrollLock /> }
```

Comment:
When using HOC `withMobileScrollLock` we wouldn't have to put this logic in here at all, which is quite disconnected from the sidebar component that needs them.